src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3414,19 +3428,87 @@ impl Coordinator {
                     // the columns refer only to referenced subsources. The best solution to
                     // this is to ensure that we simply don't try to generate the table cast
                     // in the first place.
-                    curr_references.contains(&name)
+                    if curr_subsources_by_reference.contains_key(&name) {
```

Comment:
I'm a little confused by what `name` represents in this case. Is this the 'upstream' table name or the subsource export name? If multiple subsource exports can refer to the same table name, can we clarify which this (and surrounding `*name*` variables) is referring to?

Reply:
Sure––I can change the variable name.